src/lib/utils/defer.ts

```diff
@@ -0,0 +1,19 @@
+const nextMicroTask = Promise.resolve();
+
+type Callback = (...args: any[]) => void;
+
+const defer = (callback: Callback): Callback => {
+  let progress: Promise<void> | null = null;
+  return (...args) => {
+    if (progress !== null) {
+      return;
+    }
+
+    progress = nextMicroTask.then(() => {
```

Comment:
is this not going to be shared between different calls of defer now?

Reply:
`progress` is scoped inside each call to `defer` to track whether or not we have a "work" in progress. `nextMicroTask` is shared but it doesn't matter since we only use it to defer calls to the next microtask.